src/plugins/bar-graph/chart-area.tsx

```diff
@@ -125,10 +129,16 @@ export const ChartArea = observer(function BarGraphChart({ width, height }: IPro
   }
 
   function groupedBars() {
+    // generate a unique value from the color map to force a re-render when the map changes
+    const colorMapKey = model?.secondaryAttributeColorMap?.size
+      ? JSON.stringify(getSnapshot(model.secondaryAttributeColorMap))
+      : "no-color-map";
```

Comment:
This feels hacky, but if it's the best way to do this, that's ok.

Reply:
This was the only way I could get `BarGroup` to consistently update when the colors were changed. I could be missing something, but there doesn't seem to be a less hacky option available.